### ISSUE TITLE
line 223, in get_nearest_neighbors raise NoIndexException("Index not found, please create an instance before querying")

### ISSUE THREAD

**Issue by 0xking-sol** [imartinez/privateGPT#6]
Hi, thanks for creating this incredible tool. 

I have been able to parse the .txt file correctly, and `python privateGPT.py` runs fine, outputting the "Enter a query".

But when this tries to run i get the following error message:

 line 223, in get_nearest_neighbors
    raise NoIndexException("Index not found, please create an instance before querying")

i have added two inputs of different lengths using the `python ingest.py <path_to_your_txt_file>` command.

Am i missing something simple?

**Comment by imartinez** [comment]
Apparently it is an error [thrown by Chroma db](https://github.com/chroma-core/chroma/blob/791653e5503dbb2ceb34ff37896c3c34cf6eb95a/chromadb/db/index/hnswlib.py#L230) when querying it (trying to find the context for the prompt using the vectorstore)

I'd say your **ingest process didn't work**. Re-ingest your document, and make sure it creates a 'db' folder in the root of your project, that contains an 'index' subfolder

**Comment by 0xking-sol** [comment]
I was able to fix this issue using feedback from the chromadb issue. https://github.com/imartinez/privateGPT/issues/11#issuecomment-1539159800

Basically, if you have forked this repo, update your branch to the main branch, and then reinstall requirements. 

I have since been able to run the code, ingesting and then running `python privategpt.py path/to/file`. 

I'm faced with some new issues: 
1. When i run privategpt.py, this shows up in the terminal 
'gpt_tokenize: unknown token '�'' 

2. The answer to the query seems to just return snippets of the text file rather than answering questions. 

I will play around with different methods and if i cant fix it then will open a new issue
